hack/pin-dependency.sh

```diff
@@ -58,41 +77,32 @@ trap "cleanup" EXIT SIGINT
 cleanup
 mkdir -p "${_tmp}"
 
-# Add the require directive
-echo "Running: go get ${dep}@${sha}"
-go get -d "${dep}@${sha}"
-
-# Find the resolved version
-rev=$(go mod edit -json | jq -r ".Require[] | select(.Path == \"${dep}\") | .Version")
-
-# No entry in go.mod, we must be using the natural version indirectly
-if [[ -z "${rev}" ]]; then
-  # backup the go.mod file, since go list modifies it
-  cp go.mod "${_tmp}/go.mod.bak"
-  # find the revision
-  rev=$(go list -m -json "${dep}" | jq -r .Version)
-  # restore the go.mod file
-  mv "${_tmp}/go.mod.bak" go.mod
-fi
 
-# No entry found
-if [[ -z "${rev}" ]]; then
-  echo "Could not resolve ${sha}"
-  exit 1
+# Find the resolved version before trying to use it.
+echo "Running: go mod download ${replacement}@${sha}"
+if meta=$(go mod download -json "${replacement}@${sha}"); then
+    rev=$(echo "${meta}" | jq -r ".Version")
+else
+    error=$(echo "${meta}" | jq -r ".Error")
+    echo "Download failed: ${error}" >&2
+    exit 1
 fi
+echo "Resolved to ${replacement}@${rev}"
 
-echo "Resolved to ${dep}@${rev}"
+# Add the require directive
+echo "Running: go mod edit -require ${dep}@${rev}"
```

Comment:
do all the go mod tools this work properly if $rev does not exist in $dep, but only in $replacement?

Reply:
They did when I tried this with my zapr fork. I think what happens is that the version specified in "require" gets ignored completely.

Reply:
ok, thanks for the fix